grammar/haxe.bnf

```diff
@@ -490,8 +496,16 @@ left shiftExpression ::= shiftOperator additiveExpressionWrapper
 private additiveExpressionWrapper ::= multiplicativeExpressionWrapper additiveExpression*
 left additiveExpression ::= ('+' | '-') multiplicativeExpressionWrapper
 
-private multiplicativeExpressionWrapper ::= prefixExpression multiplicativeExpression*
-left multiplicativeExpression ::= ('*' | '/' | '%') (prefixExpression | suffixExpressionWrapper)
+private multiplicativeExpressionWrapper ::= assignExpressionWrapper multiplicativeExpression*
+left multiplicativeExpression ::= ('*' | '/' | '%') assignExpressionWrapper
+
+/* Technically, assignExpressions don't have higher priority than everything else.
+ * But for parsing, this makes sense; otherwise, assignment operators that comprise
```

Comment:
It doesn't matter.  If you look at all of the other operators, they have _left_ association, which is also wrong.  This is really about being able to parse the syntax (until I can get the new parser into place).   There are a number of outstanding bugs that I'm just trying to clean up so that we don't show parse errors all over the place.

So, the reason that I needed to do this is that the parser and lexer don't communicate well, and the lexer doesn't have context to tell the difference between a type parameter ending and a shift-and-assign operator, so it tries to make the largest lexeme possible, which is the shift-and-assign.  Then, we get parser errors due to the unexpected token.  So, we don't let the lexer make larger tokens out of '>' and we try to sort it out in the parser.  

Of course, the stack of precedence rules that you see modified here tested for the shorter tokens first (because they have higher precedence).  So, we test for the assignments first.  I _could_ separate the shift-right tokens from the others and leave the others in their old position.  It just didn't seem worth it.

I spent a day and a half trying to find a way to get the parser and lexer to communicate, but it the end, there is too much isolation; there is no way for a parser external rule to get hold of the lexer to communicate state.  If it could, I could simply "enter" a type declaration and tell the lexer not to make the shift tokens until a closing bracket is found.  (I even tried to detect type names by checking for capital letters in the previous token, but that breaks because function names take type parameters, too.)  I can't just check for any identifier followed by a '<' because it's a valid operator when used with variable names.  The lexer has no idea about those and shouldn't.

Maybe there's a better way to do it that I just don't know about.  It's all going to go away soon, when I figure out how to generate IDEA's PSI and a class factory from the new parser's .mly files.  At that point, though, there's going to be a LOT of dependent code that needs to be re-written.  It may be easier to start with a clean slate...